### PR TITLE
KOGITO-5534: Codegen should use (un)marshallers for XmlElement(Ref) type

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/KOGITO-5534

In essence I now check for the presence of `@XmlElement` or `@XmlElementRef` on fields when generating the serializer\deserializer. I also do a similar check when processing the fields on beans to ensure code is also generated for classes not annotated with `@XmlMapper`.

### CODE REVIEW

File: processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/definition/BeanDefinition.java [hunk 1]
```diff
@@ -269,11 +269,6 @@ public String getQualifiedName() {
     return null;
   }
 
-  @Override
```
Comment (manstis) [inline processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/definition/BeanDefinition.java:272]: Deleted, it overrode the super class method but did nothing different.

File: processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/definition/PropertyDefinition.java [hunk 2]
```diff
@@ -149,11 +149,6 @@ public boolean isAttribute() {
     return property.getAnnotation(XmlAttribute.class) != null;
   }
 
-  @Override
```
Comment (manstis) [inline processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/definition/PropertyDefinition.java:152]: Deleted, it overrode the super class method but did nothing different.

File: processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/processor/BeanProcessor.java [hunk 3]
```diff
@@ -90,7 +90,10 @@ private void processBean(TypeElement bean) {
 
   private void processField(VariableElement field) {
     if (checkField(field)) {
-      checkTypeAndAdd(field.asType());
+      // Ensure the serializer/deserializer is generated for the fields concrete type, considering
```
Comment (manstis) [inline processor/src/main/java/org/treblereel/gwt/xml/mapper/apt/processor/BeanProcessor.java:93]: Don't assume the field's type is correct.. try to look it up from annotations too.

**Comment by treblereel** [comment]
@manstis very cool, thanks a million !

**Comment by manstis** [comment]
@treblereel Happy to help... it'd be great if you'd be able to release a newer version (with this fix) at some point.

**Comment by treblereel** [comment]
@manstis i think, i can switch jaxb to jakartaee and release all together, wdyt ?

**Comment by manstis** [comment]
@treblereel You could... but once the `mapper-xml` plugin is upgraded everyone that uses it will need to move too.

I'm more than happy to move my work to Jakarta however IDK if the DMN or BPMN teams would be affected?

If they are, then the move to Jakarta probably needs to be coordinated; or at least communicated.

**Comment by treblereel** [comment]
@manstis ok, let's wain for Roger...

**Comment by manstis** [comment]
@treblereel I'll send an email tomorrow.

**Pull request merged**